src/java/org/apache/cassandra/config/Config.java

```diff
@@ -903,11 +903,18 @@ public static void setClientMode(boolean clientMode)
     public volatile boolean zero_ttl_on_twcs_warned = true;
     public volatile boolean zero_ttl_on_twcs_enabled = true;
     public volatile boolean non_partition_restricted_index_query_enabled = true;
-    public volatile int sai_sstable_indexes_per_query_warn_threshold = 32;
-    public volatile int sai_sstable_indexes_per_query_fail_threshold = -1;
     public volatile boolean intersect_filtering_query_warned = true;
     public volatile boolean intersect_filtering_query_enabled = true;
 
+    public volatile int sai_sstable_indexes_per_query_warn_threshold = 32;
+    public volatile int sai_sstable_indexes_per_query_fail_threshold = -1;
+    public volatile DataStorageSpec.LongBytesBound sai_string_term_size_warn_threshold = new DataStorageSpec.LongBytesBound("1KiB");
```

Comment:
@maedhroz you should also update `cassandra.yaml` with all these guardrails otherwise people wont have a clue they are available.

Reply:
by the way ... where are all these values for defaults coming from? I am not necessarily arguing the sizes as such, I am curious based on what we put these sizes there? Is that some kind of an empirical observation or something else?

Reply:
So you'll notice the frozen and vector fail thresholds are just the previous system property maxes: https://github.com/apache/cassandra/pull/3241/files#diff-32a8b4fa02cda1ef5a7216fe68d70d23a9784a4bb33b1230d2ec776140dae3d7L428

For strings, I bumped the fail threshold to a number @JeremiahDJordan had mentioned in Slack that Astra was using as a fail threshold (and left the previous max as the new warning threshold).

Reply:
I would set frozen and string to be the same at 8k.
I think 32k still works for vectors.  4096 float32 is still the largest embedding I have seen truly used.

Reply:
So how does this look?

string: 1KiB warn/8KiB fail
frozen: 1KiB warn/8KiB fail
vector: 16KiB warn/32KiB fail

Reply:
I don't have an opinion here. I don't have production numbers to refer to neither I know SAI in that regard, so whatever you guys think it's reasonable.

Reply:
I dont know if we should go with the defaults in the first place. That is my biggest concern. I would just keep it disabled and people might opt-in. You can bring this to ML and ask. Maybe I am a little bit paranoid here but we should triple check that these figures make sense to as broad audience as possible.

Reply:
I think @JeremiahDJordan's input counts for something here, given he's the only one of us running SAI in production (in Astra). Also, if you look at this patch from the perspective of what was here before, all I've done is take the same (or even more lenient) max term sizes, and make sure writes fail visibly for the client. This seems strictly better than silently compromising the index with not so much as a warning.

The limits on term size before my patch here were derived at some point during development at DS to protect the index, and I'm -100 on them being opt-in, especially because they are brand new in 5.0 (as is SAI itself). I'm more or less willing to die on this hill.

CC @mike-tr-adamson

Reply:
Alright. Let's keep it like it is now.